index.d.ts

```diff
@@ -0,0 +1,68 @@
+// Version-loose types for the `history` methods used by this module
+// - These must be updated for newer versions of `history` and as more methods are used
+type Location = {
+  path: string;
+} | {
+  pathname: string;
+  search: string;
+}
+interface History3 {
+  getCurrentLocation(): Location;
+}
+interface History4 {
+  location: Location;
+}
+type LooseHistory = {
+  listen(listener: (location: Location, ...unknownArgs: unknown[]) => void): void;
+} & (History3 | History4);
+
+/**
+ * Options for the `Piwik`/`ReactPiwik` constructor
+ */
+type PiwikOptions = {
+  url: string;
+  siteId: number;
+  jsFilename?: string;
+  phpFilename?: string;
+  enableLinkTracking?: boolean;
+  trackDocumentTitle?: boolean;
+  pathBasename?: string;
```

Comment:
How would you like to handle `trackErrors`? It doesn't do anything but is [mentioned in the readme's example](https://github.com/guillaumeparis2000/react-piwik#usage).

Reply:
It's something that I want to implement but I never have the time to do it, I think it will be better to remove it from here and from the readme.

Reply:
Removed from the readme and didn't add to the types.